### PR TITLE
feat(KIcon) update docs, remove unused icons

### PR DESCRIPTION
**Pull request by travispamaral**
### Summary
Adds updated KIcon doc and contributing docs. Removes unused icons and fixes KIcon viewbox not being applied correctly.

fixes https://github.com/Kong/kongponents/issues/247
closes https://github.com/Kong/kongponents/issues/251

#### Changes made:
* Updates default size and viewbox values
* Removes unused/old icons
* Updates icon docs
* Updates contributing docs

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

**Comment by xitongzou** [comment]
Tested this out on KHCP which looks good but you might want to look at Kong Manager as well. It does use KIcon after all unless we plan on never upgrading that.

File: docs/components/icon.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Icon
 
-**KIcon** - Icon component for displaying SVG symbols.
+**KIcon** - Icon component for displaying SVG symbols. Currently Size, color and viewbox vary between icons, so the implementor must be mindful when using.
```
Comment (kaiarrowood) [inline docs/components/icon.md:3]: Did you maybe want to mention how to update/view the docs to verify that the icon is rendering correctly? Or is that too obvious?

File: docs/contributing/adding-icons-to-kicon.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+1) **Update Tests**  
+The last step is to ensure the `KIcon` tests pass. Each icon is tested to ensure it matches an existing snapshot. Because this is a new icon you will need to update the snapshots. From The root directory run `yarn test packages/KIcon/KIcon.spec.js --updateSnapshot`.
+
+Thats it you're done! Git push and PR! :tada:
+
+::: warning
+KIcon can only contain path elements. If the icon you are bringing in includes circle, ellipse, rect, text, etc those elements will not be displayed. It is recommended to update your icon to only paths before bringing into Kongponents.
+:::
+
+<!-- Add file to KIcon/icons with name format icn-<icon_name>.svg
```
Comment (kaiarrowood) [inline docs/contributing/adding-icons-to-kicon.md:34]: Did you maybe want to mention how to update/view the docs to verify that the icon is rendering correctly? Or is that too obvious?

**Comment by travispamaral** [comment]
> Tested this out on KHCP which looks good but you might want to look at Kong Manager as well. It does use KIcon after all unless we plan on never upgrading that. 

I took a quick look at a few spots but can't remember if it was KM or KHCP. I'll check again but nothing should be changed. This update is backward compatible. I'll check before merging

File: docs/contributing/adding-icons-to-kicon.md (hunk 2 above)
Comment (travispamaral) [reply docs/contributing/adding-icons-to-kicon.md:34]: I think it's not needed as we have docs earlier on explaining how to start the app.

**Pull request merged**